pytd/tests/test_writer.py

```diff
@@ -273,22 +270,22 @@ def test_write_dataframe_tempfile_deletion(self):
                 pd.DataFrame([[1, 2], [3, 4]]), self.table, "overwrite"
             )
         fp = self.writer._bulk_import.call_args[0][1]
-        self.assertFalse(os.path.isfile(fp.name))
+        self.assertFalse(os.path.isfile(fp[0].name))
 
     def test_write_dataframe_msgpack(self):
         df = pd.DataFrame([[1, 2], [3, 4]])
         self.writer.write_dataframe(df, self.table, "overwrite", fmt="msgpack")
         api_client = self.table.client.api_client
         self.assertTrue(api_client.create_bulk_import.called)
         self.assertTrue(api_client.create_bulk_import().upload_part.called)
-        _bytes = BulkImportWriter()._write_msgpack_stream(
-            df.to_dict(orient="records"), io.BytesIO()
-        )
-        size = _bytes.getbuffer().nbytes
-        api_client.create_bulk_import().upload_part.assert_called_with(
-            "part", ANY, size
-        )
-        self.assertFalse(api_client.create_bulk_import().upload_file.called)
+        with tempfile.NamedTemporaryFile(delete=False) as fp:
+            fp = BulkImportWriter()._write_msgpack_stream(
+                df.to_dict(orient="records"), fp
+            )
+            api_client.create_bulk_import().upload_part.assert_called_with(
+                "part-0", ANY, 62
```

Comment:
maybe I missed something, but where this `62` comes from?

Reply:
I tried to get the file size from `fp`, but I gave up because it doesn't match with the actual file size in the `_write_msgpack_stream()` method.